### ISSUE TITLE
[Bug]: node_modules cause copy_to_bin performance regression when passing between ts_project and npm_package

### ISSUE THREAD

**Issue by Aghassi** [aspect-build/rules_ts#367]
### What happened?

In our code setup, we have `ts_project` for source code, and `ts_project` for test code. That means we will needs `@types` packages as well as `jest-` and `@jest` packages to pass typechecking. While we do have `include_runfiles = False` in our `npm_package`, there is still a massive list of node_modules generated from `ts_project` to the `<target.json>` file that is in the `bazel-bin` folder. We should not be forwarding packages that are used for typechecking only, we should only forward dependencies the code needs to run.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`:
6.2.0

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
`rules_ts` - 1.0.1

Language(s) and/or frameworks involved:

### How to reproduce

```shell
It's probably best to reach out to use directly, but a basic example is if you do something like 


ts_project(
  name,
  srcs = ['index.ts'],
  data = ['//:node_modules/@types']
)
```

This is an exaggeration, but on a reasonably large pnpm node_modules tree, this can generate a multi meg json payload (ours is 3.2mb)
```


### Any other information?

_No response_

**Comment by Aghassi** [comment]
Looks related to this too https://github.com/aspect-build/rules_js/issues/1032

**Comment by Aghassi** [comment]
We did some more debugging with @jbedard today. We found that our perf regression was directly correlated with `npm_package` having `include_transitive_declarations` default to `True`. When we flipped that to `False`, our copy_to_bin was eliminated and the underlying config shrunk from 3.2mb to 8kb.

This works for us because we use `skipLibCheck` and don't expect nested node_modules to forward types. The package we are building will forward types and source files, that is it.
